cliquet/resource.py

```diff
@@ -330,9 +330,8 @@ def update_record(self, record, old=None):
             if nothing_changed:
                 return new
 
-        record_id = new[self.id_field]
         try:
-            return self.db.update(record_id=record_id,
+            return self.db.update(record_id=self.record_id,
```

Comment:
Are we sure `self.record_id == new[self.id_field]` ? We're kinda bypassing the information provided in the `new` object if we're looking at what's in `self.record_id`.